src/tree.ts

```diff
@@ -71,38 +71,51 @@ export class Tree {
     return node;
   }
 
-  setter(index: Gindex, expand = false): Link {
-    let link = identity;
+  setNode(index: Gindex, n: Node, expand = false): void {
     let node = this.rootNode;
-    const iterator = gindexIterator(index);
-    for (const i of iterator) {
-      if (i) {
-        if (node.isLeaf()) {
-          if (!expand) throw new Error(ERR_INVALID_TREE);
-          else {
-            const child = zeroNode(iterator.remainingBitLength() - 1);
-            node = new BranchNode(child, child);
-          }
+
+    // Pre-compute entire bitstring instead of using an iterator (25% faster)
+    const bitstring = index.toString(2);
```

Comment:
Can you check the index < 1 case and add unit test? Previously handled by gindexIterator